File: demo/ex_2.cpp
/*  Example 2: Ownership

    Compile ->  g++ ex_2.cpp
    Run ->      ./a.out
*/

#include <iostream>
#include <string>

int main()
{
    std::string string_1 = "Stay Froggy!";
    std::string string_2 = string_1;

    std::cout << string_1 << std::endl;
    std::cout << string_2 << std::endl;

}

File: demo/ex_2.rs
/*  Example 2: Ownership

    Compile ->  rustc ex_2.rs
    Run ->      ./ex_2
*/

fn main() {
    let string_1:String = "Stay Froggy!".to_string();
    let string_2:String = string_1;  // Compiler won't let you do this

    println!("{}", string_1);
    println!("{}", string_2);  // Nor this
}